### ISSUE TITLE
Pickling of sklearn.tree.Tree

### ISSUE THREAD

**Issue by pyodave** [pyodide/pyodide#521]
When running the code below the line (note I am pickling/unpickling directly to a byte string and not writing a file object, the same error occurs with a file object pointing to a io buffer).
```py
reLoadCLF = pickle.loads(pickle.dumps(clf))
```

crashes with   File "sklearn/tree/_tree.pyx", line 601, in 
```py
sklearn.tree._tree.Tree.__cinit__
ValueError: Buffer dtype mismatch, expected 'SIZE_t' but got 'long long'
```

I think it should work since we aren't running a 64 vs a 32 bit Python nor having different library versions. I have tried using different protocols and different skLearn classifiers to no avail. Is there a way to do this with the existing codebase or is this a bug?
```py
#Random forest classsifier
from sklearn.ensemble import RandomForestClassifier
from sklearn.datasets import load_iris
from sklearn.model_selection import train_test_split
from sklearn.metrics import accuracy_score
#import pandas as pd

#loading the dataset
iris = load_iris()
X = iris.data
print(X)
y = iris.target
print(y)

#Split the data
X_train, X_test, y_train, y_test = train_test_split(X,y,random_state=42,test_size=0.5)

#Build the model
clf = RandomForestClassifier(n_estimators=10)

#Train the classifier
clf.fit(X_train, y_train)

#Predictions
predicted = clf.predict(X_test)

#Check accuracy
print(accuracy_score(predicted, y_test))

import pickle
reLoadCLF = pickle.loads(pickle.dumps(clf))
print('reLoad & go')
predictedRL = reLoadCLF.predict(X_test)
print(accuracy_score(predictedRL, y_test))
```

**Comment by rth** [comment]
Thanks for the report. 

Pickling should work in general, this is a problem specifically with [`sklearn.tree._tree.Tree`](https://github.com/scikit-learn/scikit-learn/blob/9caf8350e7cf09fa3d59c8f34758f0dae819398c/sklearn/tree/_tree.pyx#L607). My guess is [some of the type conversions](https://github.com/scikit-learn/scikit-learn/blob/9caf8350e7cf09fa3d59c8f34758f0dae819398c/sklearn/tree/_utils.pyx#L57) done there do not work as expected in WASM environement. In pariticular it sounds like [`Tree.n_classes`](https://github.com/scikit-learn/scikit-learn/blob/master/sklearn/tree/_tree.pyx#L567) returned a `long long` while it should have been `size_t` which is `unsigned long` in emscripten (cf https://github.com/emscripten-core/emscripten/pull/5916)

**Comment by rth** [comment]
This was resolved in the future scikit-learn 1.1.0 release (scikit-learn/scikit-learn#19602), however we are currently blocked by scipy version #549 for updating.

**Comment by rth** [comment]
I confirm the provided example works in the latest version of Pyodide. Closing as resolved.
